app/src/main/java/io/github/hidroh/materialistic/WebFragment.java

```diff
@@ -288,7 +292,31 @@ protected void load() {
 
     private void loadUrl() {
         setWebSettings(true);
-        mWebView.reloadUrl(mItem.getUrl());
+        reloadUrl(mItem.getUrl());
+    }
+
+    private void reloadUrl(String url) {
+        reloadUrl(url, null);
+    }
+
+    @SuppressLint("AddJavascriptInterface")
+    private void reloadUrl(String url, @Nullable String pdfFilePath) {
+        if (mPdfAndroidJavascriptBridge != null) {
+            mPdfAndroidJavascriptBridge.cleanUp();
```

Comment:
Shouldn't this be `finalize()` instead?

Reply:
Also we should call it in `onDestroy()` as well, or else we only clean up the previously opened file when we need to load a new one.

Reply:
I thought it's considered a bad practice to call `finallize()` explicitly, and the GC gonna call it whenever it needs to. And here I want to clean it up before GC, because we don't really need the opened PDF file for the previous page anymore.

Reply:
Ah ok got it, I forgot that `finalize()` is destructor in Java, so rarely use it. But if we call `cleanUp()` in `onDestroy()` it's more in our control to clean up when we close the article, you don't want to wait until it hits GC.